src/app/GitUI/CommandsDialogs/FormVerify.cs

```diff
@@ -312,12 +341,49 @@ private void UpdateLostObjects()
                         .WhereNotNull()
                         .OrderByDescending(l => l.Date));
 
+                LostObject[] commits = _lostObjects.Where(o => o.ObjectType == LostObjectType.Commit).ToArray();
+                List<string> metadata = new(commits.Length);
+                int batchSize = 30_000 / (ObjectId.Sha1CharCount + 1); // Based on process max command line length and hash length (with a margin)
+                int currentBatch = 0;
+
+                while (currentBatch * batchSize < commits.Length)
+                {
+                    LostObject[] nextBatch = commits.Skip(currentBatch * batchSize)
+                        .Take(batchSize).ToArray();
+                    currentBatch++;
+                    string[] metadataBatch = LostObject.GetCommitsMetadata(Module, string.Join(" ", nextBatch.Select(c => c.ObjectId.ToString())));
+                    metadata.AddRange(metadataBatch);
+                }
+
+                for (int i = 0; i < commits.Length; i++)
+                {
+                    commits[i].FillCommitData(Module, metadata[i]);
+                }
+
                 UpdateFilteredLostObjects();
             }
         }
 
+        private bool _typeDetected = false;
```

Comment:
Please keep fields at the top. 
Initialisation is not required, it gets initialised implicitly.